addons/website/static/src/js/editor/widget_link.js

```diff
@@ -48,29 +53,81 @@ weWidgets.LinkTools.include({
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    _isNewWindow: function (url) {
+        // If URL starts with /web/ or if the file name is included in URL, then it's a file.
+        // Thus, we want to open it in a new window.
+        const fileName = this.$link.attr('data-file-name') && this.$link.attr('data-file-name').replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+        const isFileLink = new RegExp(`${fileName ? fileName + '|' : ''}\\/web\\/`).test(url);
+        // If a link that pointed to a file before doesn't anymore, we need to update the toolbar.
+        if (!isFileLink && fileName) {
+            this.$link.removeAttr('data-file-name');
+            $('.o_we_link_replace_file').parent().addClass('d-none');
+            $('.o_we_link_upload_file').removeClass('d-none');
+        }
+        return isFileLink;
+    },
+    /**
+     * @private
+     */
+    _onLinkUploadFileClick: function () {
+        const documentDialog = new weWidgets.MediaDialog(this, {
+            noIcons: true,
+            noImages: true,
+            noVideos: true,
+        });
+        documentDialog.on('save', this, function (htmlOutput) {
+            const $elem = $(htmlOutput);
+            if ($elem.length) {
+                const url = $elem.is('a') ? $elem.attr('href').replace('&download=true', '') : $elem.attr('src');
+                this.$urlInput.val(url).trigger('input');
+                // We need to save the file name in order to display it in the toolbar.
+                const file = documentDialog.activeWidget.selectedAttachments[0];
+                if (file) {
+                    this.$link.attr('data-file-name', file.name);
+                }
+            }
+        });
+        documentDialog.on('closed', this, function () {
+            // Link tool is destroy when we click outside the toolbar (eg. the dialog)
+            // So we need to open it silently after dialog closure.
+            this.$link.click();
+            this.$button.click();
+        });
+        documentDialog.open();
```

Comment:
Seems to serve the same purpose as my other comment and thus for the wrong reasons :wink: The links tools should just not be closed during dialog use, so there would be no need to do hacky stuff like here :wink: